packages/ubuntu_wsl_setup/lib/pages/installation_slides/installation_slides_page.dart

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:subiquity_client/subiquity_client.dart';
+import 'package:ubuntu_service/ubuntu_service.dart';
+import 'package:ubuntu_widgets/ubuntu_widgets.dart';
+import 'package:ubuntu_wizard/widgets.dart';
+import 'package:ubuntu_wsl_setup/l10n.dart';
+import 'package:yaru_icons/yaru_icons.dart';
+
+import '../../constants.dart';
+import '../../l10n/app_localizations.dart';
+import '../../services/journal.dart';
+import 'slides.dart';
+import 'installation_slides_model.dart';
+
+/// Implements a slide show in which all slides have a common background image,
+/// meant to only exist while Subiquity server is not up and running, (like
+/// when WSL is still registering the distro).
+/// Control comes from the SlideShow widget from ubuntu_widgets package.
+class InstallationSlidesPage extends StatefulWidget {
+  const InstallationSlidesPage({Key? key}) : super(key: key);
+
+  @override
+  State<InstallationSlidesPage> createState() => _InstallationSlidesPageState();
+
+  static Widget create(BuildContext context) {
+    final slides = theSlides(context);
+    return ChangeNotifierProvider(
+      create: (_) => InstallationSlidesModel(
+        journal: getService<JournalService>().stream,
+        monitor: getService<SubiquityStatusMonitor>(),
+      ),
+      child: Provider<SlidesProvider>(
+          create: (context) => SlidesProvider(slides),
+          child: const InstallationSlidesPage()),
+    );
+  }
+}
+
+class _InstallationSlidesPageState extends State<InstallationSlidesPage> {
+  @override
+  void initState() {
+    super.initState();
+    final model = context.read<InstallationSlidesModel>();
+    model.init(onServerUp: Wizard.of(context).next);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final model = context.read<InstallationSlidesModel>();
```

Comment:
should it be `context.watch`?

Reply:
Indeed. I also considering that I could watch on the SlidesProvider as well and use it to display different error screens (slides)... But that's topic for another PR. This one is already bigger than I originally anticipated.